src/GamecubeAPI.hpp

```diff
@@ -87,21 +87,40 @@ bool CGamecubeController::read(void)
         }
     }
 
-    // Read the controller, abort if it fails.
-    // Additional information: If you press X + Y + Start on the controller for 3 seconds
-    // It will turn off unless you release the buttons. The recalibration is all done
-    // on the Gamecube side. If the controller resets origin will have different values for sure.
-    if (!gc_read(pin, &report, status.rumble))
+
+    if (status.device == NINTENDO_DEVICE_GC_WIRED)
     {
-        reset();
-        return false;
-    }
+        // Read the controller, abort if it fails.
+        // Additional information: If you press X + Y + Start on the controller for 3 seconds
+        // It will turn off unless you release the buttons. The recalibration is all done
+        // on the Gamecube side. If the controller resets origin will have different values for sure.
+        if (!gc_read(pin, &report, status.rumble))
+        {
+            reset();
+            return false;
+        }
 
-    // Check if controller reported that we read the origin values (check if it disconnected).
-    // The Gamecube would just request (instantly) the origin again, but we keep things simple.
-    if (report.origin) {
-        reset();
-        return false;
+        // Check if controller reported that we read the origin values (check if it disconnected).
+        // The Gamecube would just request (instantly) the origin again, but we keep things simple.
+        if (report.origin) {
+            reset();
+            return false;
+        }
```

Comment:
Do we need this for the keyboard as well?

Reply:
From what I can tell, no.

https://www.raphnet.net/electronique/gc_n64_usb/index_en.php#8

This is where I got most of my information. The keyboard will just initialize and function as soon as the 0x540000  command is given.